client/lib/emails/can-add-mailboxes-to-domain.ts

```diff
@@ -0,0 +1,19 @@
+import { canCurrentUserAddEmail } from 'calypso/lib/domains';
+import { hasEmailForwards } from 'calypso/lib/domains/email-forwarding';
+import { getGSuiteProductSlug, getGSuiteSubscriptionStatus } from 'calypso/lib/gsuite';
+import { getTitanProductSlug } from 'calypso/lib/titan';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+/**
+ * Determines whether a domain has email provider and is everything set up to add new mailboxes
+ */
+export function canAddMailboxesToDomain( domain?: ResponseDomain ): boolean {
+	const domainIsNotProvisioning = getGSuiteProductSlug( domain ) || getTitanProductSlug( domain );
+	const isAwaitingGoogleTosAcceptance = getGSuiteSubscriptionStatus( domain ) === 'suspended';
+
+	return (
+		( domainIsNotProvisioning || hasEmailForwards( domain ) ) &&
```

Comment:
```suggestion
	const emailIsProvisioning = ! getGSuiteProductSlug( domain ) && ! getTitanProductSlug( domain );
	const isAwaitingGoogleTosAcceptance = getGSuiteSubscriptionStatus( domain ) === 'suspended';

	return (
		( ! emailIsProvisioning || hasEmailForwards( domain ) ) &&
```
The naming here came from me in a hurried PR. I think we'd be better served to change this accordingly!

Reply:
I think naming can also be improved in the name of the method itself. Technically mailboxes are not added to a domain, they are added to the Email Subscription.

What about `canAddMailboxesToEmailSubscription`